src/org/omegat/core/data/ExternalTMFactory.java

```diff
@@ -118,26 +123,45 @@ public ExternalTMX load(Language sourceLang, Language targetLang) throws Excepti
             TMXReader2.LoadCallback loader = new TMXReader2.LoadCallback() {
                 public boolean onEntry(TMXReader2.ParsedTu tu, TMXReader2.ParsedTuv tuvSource,
                         TMXReader2.ParsedTuv tuvTarget, boolean isParagraphSegtype) {
+
                     if (tuvSource == null) {
                         return false;
                     }
 
+                    // Keep all the Tuvs matching at least the target language
+                    for (TMXReader2.ParsedTuv tuv : tu.tuvs) {
+                        String[] tuvLangParts = tuv.lang.split("\\W");
+                        if (!targetLang.getLanguageCode().equalsIgnoreCase(tuvLangParts[0])) {
```

Comment:
This ad hoc matching strategy seems a bit dubious. Why not use `Language` to compare?

Reply:
You are correct, it's not the most elegant solution. I wanted to maintain the same behaviour used in `org.omegat.util.TMXReader2.getTuvByLang`. 
A better approach would be to add a method like `Language.isSimilar(String)` to check if two locales share the same language.

Reply:
There are two new methods in Language `isSameCountryLanguage` and `isSameLanguage` to abstract these feature.